packages/canon/src/components/TextField/TextField.stories.tsx

```diff
@@ -38,137 +40,107 @@ type Story = StoryObj<typeof meta>;
   args: {
     ...Default.args,
-    label: 'Label',
-    description: 'Description',
-    required: true,
   },
   render: args => (
     <Flex direction="row" gap="4" style={{ width: '100%', maxWidth: '600px' }}>
-      <TextField {...args} labelSize="small" />
-      <TextField {...args} labelSize="medium" />
+      <TextField {...args} size="small" icon={<Icon name="sparkling" />} />
+      <TextField {...args} size="medium" icon={<Icon name="sparkling" />} />
     </Flex>
   ),
 };
 
-export const HideLabelAndDescription: Story = {
+export const DefaultValue: Story = {
   args: {
-    ...WithLabel.args,
-    hideLabelAndDescription: true,
-  },
-};
-
-export const Disabled: Story = {
-  args: {
-    ...WithLabel.args,
-    disabled: true,
+    ...Default.args,
+    defaultValue: 'https://example.com',
   },
 };
 
-export const Sizes: Story = {
+export const WithLabel: Story = {
   args: {
     ...Default.args,
     label: 'Label',
-    description: 'Description',
   },
-  render: args => (
-    <Flex direction="row" gap="4" style={{ width: '100%', maxWidth: '600px' }}>
-      <TextField {...args} size="small" icon={<Icon name="sparkling" />} />
-      <TextField {...args} size="medium" icon={<Icon name="sparkling" />} />
-    </Flex>
-  ),
 };
 
-export const Responsive: Story = {
+export const WithDescription: Story = {
   args: {
     ...WithLabel.args,
-    size: {
-      initial: 'small',
-      sm: 'medium',
-    },
+    description: 'Description',
   },
 };
 
-export const WithError: Story = {
+export const Required: Story = {
   args: {
     ...WithLabel.args,
-    error: 'Invalid URL',
+    isRequired: true,
   },
 };
 
-export const WithErrorAndDescription: Story = {
+export const Disabled: Story = {
   args: {
-    ...WithLabel.args,
-    error: 'Invalid URL',
-    description: 'Description',
+    ...Default.args,
+    isDisabled: true,
   },
 };
 
 export const WithIcon: Story = {
   args: {
-    ...WithLabel.args,
-    placeholder: 'Search...',
+    ...Default.args,
+    defaultValue: 'Search...',
     icon: <Icon name="search" />,
   },
 };
 
 export const DisabledWithIcon: Story = {
   args: {
     ...WithIcon.args,
-    disabled: true,
+    isDisabled: true,
   },
 };
 
-export const WithOnClear: Story = {
+export const ShowError: Story = {
   args: {
     ...WithLabel.args,
-    placeholder: 'Search...',
-    type: 'search',
-    onClear: () => console.log('Cleared!'),
   },
+  render: args => (
+    <Form validationErrors={{ url: 'Invalid URL' }}>
+      <TextField {...args} />
+    </Form>
+  ),
 };
 
-export const DisabledWithOnClear: Story = {
+export const Validation: Story = {
   args: {
-    ...WithOnClear.args,
-    defaultValue: 'Testing',
-    disabled: true,
-  },
+    ...WithLabel.args,
+    validate: value => (value === 'admin' ? 'Nice try!' : null),
+  },
+};
+
+export const CustomField: Story = {
+  render: () => (
+    <>
+      <FieldLabel
+        htmlFor="custom-field"
+        id="custom-field-label"
```

Comment:
Thinking we may want to encourage `useId`? although we do still aim to support React 17 for the time being so a bit of a dilemma.